pkg/virtctl/credentials/add-key/add-key_test.go

```diff
@@ -0,0 +1,562 @@
+
+		// VM is not running
+		vmi = nil
+
+		err := runAddKeyCommand(
+			"--user", newUser,
+			"--value", testKey,
+			vmName,
+		)
+		Expect(err).ToNot(HaveOccurred())
+
+		var newSecretName string
+		for _, credential := range vm.Spec.Template.Spec.AccessCredentials {
+			if credential.SSHPublicKey != nil &&
+				credential.SSHPublicKey.PropagationMethod.QemuGuestAgent != nil {
+				for _, user := range credential.SSHPublicKey.PropagationMethod.QemuGuestAgent.Users {
+					if user == newUser {
+						newSecretName = credential.SSHPublicKey.Source.Secret.SecretName
+					}
+				}
+			}
+		}
+
+		Expect(newSecretName).ToNot(BeEmpty())
+
+		secret, err := kubeClient.CoreV1().Secrets(metav1.NamespaceDefault).Get(context.Background(), newSecretName, metav1.GetOptions{})
+		Expect(err).ToNot(HaveOccurred())
+
+		Expect(secret.Data).To(HaveLen(1))
+		Expect(secret.Data).To(ContainElement([]byte(testKey)))
+	})
+
+	It("should fail if multiple secrets are specified for a user, and --secret parameter is not set", func() {
+		vm.Spec.Template.Spec.AccessCredentials = append(vm.Spec.Template.Spec.AccessCredentials,
+			v1.AccessCredential{
+				SSHPublicKey: &v1.SSHPublicKeyAccessCredential{
+					Source: v1.SSHPublicKeyAccessCredentialSource{
+						Secret: &v1.AccessCredentialSecretSource{
+							SecretName: "second-secret",
+						},
+					},
+					PropagationMethod: v1.SSHPublicKeyAccessCredentialPropagationMethod{
+						QemuGuestAgent: &v1.QemuGuestAgentSSHPublicKeyAccessCredentialPropagation{
+							Users: []string{userName},
+						},
+					},
+				}})
+
+		err := runAddKeyCommand(
+			"--user", userName,
+			"--value", "test-key",
+			vmName,
+		)
+		Expect(err).To(MatchError(ContainSubstring("multiple secrets specified")))
+	})
+
+	It("should patch secret", func() {
+		const testKey = "test-key"
+		err := runAddKeyCommand(
+			"--user", userName,
+			"--value", testKey,
+			vmName,
+		)
+		Expect(err).ToNot(HaveOccurred())
+
+		secret, err := kubeClient.CoreV1().Secrets(metav1.NamespaceDefault).Get(context.Background(), secretName, metav1.GetOptions{})
+		Expect(err).ToNot(HaveOccurred())
+
+		Expect(secret.Data).To(HaveLen(1))
+		Expect(secret.Data).To(ContainElement([]byte(testKey)))
+	})
+
+	It("should add second key to secret", func() {
+		secret, err := kubeClient.CoreV1().Secrets(metav1.NamespaceDefault).Get(context.Background(), secretName, metav1.GetOptions{})
+		Expect(err).ToNot(HaveOccurred())
+
+		secret.Data = map[string][]byte{
+			"old-key.pub": []byte("old-key"),
+		}
+
+		_, err = kubeClient.CoreV1().Secrets(metav1.NamespaceDefault).Update(context.Background(), secret, metav1.UpdateOptions{})
+		Expect(err).ToNot(HaveOccurred())
+
+		const testKey = "test-key"
+		err = runAddKeyCommand(
+			"--user", userName,
+			"--value", testKey,
+			vmName,
+		)
+		Expect(err).ToNot(HaveOccurred())
+
+		secret, err = kubeClient.CoreV1().Secrets(metav1.NamespaceDefault).Get(context.Background(), secretName, metav1.GetOptions{})
```

Comment:
Can we include this creation and error check in the function `createSecret` and use this when possible? It will reduce the test size

Reply:
done